dallinger/experiment_server/experiment_server.py

```diff
@@ -742,35 +742,28 @@ def create_participant(worker_id, hit_id, assignment_id, mode):
         + 1
     )
 
-    recruiter_name = request.args.get("recruiter", "undefined")
-    if not recruiter_name or recruiter_name == "undefined":
-        recruiter = recruiters.from_config(_config())
-        if recruiter:
-            recruiter_name = recruiter.nickname
+    recruiter_name = request.args.get("recruiter")
 
     # Create the new participant.
-    participant = models.Participant(
-        recruiter_id=recruiter_name,
-        worker_id=worker_id,
-        assignment_id=assignment_id,
-        hit_id=hit_id,
-        mode=mode,
-        fingerprint_hash=fingerprint_hash,
-    )
-
     exp = Experiment(session)
+    participant = exp.create_participant(
+        worker_id, hit_id, assignment_id, mode, recruiter_name, fingerprint_hash
+    )
 
+    session.flush()
     overrecruited = exp.is_overrecruited(nonfailed_count)
```

Comment:
I feel like this back-and-forth is weird (it's always and is just more obvious now) and the status should just be set directly in `create_participant`... could we pass `nonfailed_count` to that method?

Reply:
I started out by moving this status check and related queries into `create_participant`, but eventually the `quorum` data that's needed in the route output made it a bit unwieldy. It seemed like we'd have to either do a some calculations twice or pass additional parameters to `Experiment.create_participant`, since that method should probably return a `Participant` object rather than some related data structure. As a result I decided to keep the experiment method basic.

Reply:
That makes sense. Sounds like the right move.